### ISSUE TITLE
[Bug] Scheduled workflow did not receive OpenTelemetry tracing

### ISSUE THREAD

**Issue by Holi0317** [temporalio/sdk-python#449]
### What are you really trying to do?

Implement otel tracing for our workflows, where most are triggered by temporal schedule.

### Describe the bug

Trace information is not generated and propagated for scheduled workflow executions.

### Minimal Reproduction

https://github.com/Holi0317/temporal-bugreport/tree/py-schedule-tracing

1. Run `poetry install`
2. Start jaeger with following snippet:
```bash
docker run -d --name jaeger \
  -e COLLECTOR_OTLP_ENABLED=true \
  -p 16686:16686 \
  -p 4317:4317 \
  -p 4318:4318 \
  jaegertracing/all-in-one:latest
```
3. Run `poetry run python temporal_tracing/worker.py` to start the worker
4. Run `poetry run python temporal_tracing/starter.py` to run the workflow once by manual invoke
5. Run `poetry run python temporal_tracing/scheduler.py` to schedule workflow execution. Wait til a minute passed and there should be execution result of scheduled execution

If the workflow is scheduled, jaeger shows 2 distinct traces with different trace ID shown in the console, each only with `RunActivity:compose_greeting` in it

```
Worker started, ctrl+c to exit
Hello in activity. Let's check the traces b69e29b63117d84f98d1095948520fd3 55c467efaf77c85d
Hello in activity. Let's check the traces daea5868fe257f559621af68d9bd869d 66f6fc4a8d938155
```

![SCR-20231214-tpaa](https://github.com/temporalio/sdk-python/assets/6838982/d5d77ce5-4eb2-4791-9576-22d4c7fef514)

If the workflow is invoked manually, jaeger shows a single trace that looks not broken. And the trace ID are the same across activities:

```
Worker started, ctrl+c to exit
Hello in activity. Let's check the traces 3fd71755262b64b3bc6097d93f419822 cb8aca4ed13cfc11
Hello in activity. Let's check the traces 3fd71755262b64b3bc6097d93f419822 598ecfcfb4157a7e
```

![SCR-20231214-tqpo](https://github.com/temporalio/sdk-python/assets/6838982/4376c7e0-514f-4973-825b-9316b94998e0)
![SCR-20231214-tqqa](https://github.com/temporalio/sdk-python/assets/6838982/94aefb43-2a1b-4706-ae47-0e29a742df2a)

### Environment/Versions

- OS and processor: M1 Mac
- Temporal Version: [e.g. 1.14.0?] and/or SDK version: Temporal version 0.10.7 (server 1.22.2) (ui 2.21.3). SDK 1.4.0
- Are you using Docker or Kubernetes or building Temporal from source? Temporal dev server from CLI

### Additional context

Kinda relates to https://github.com/temporalio/sdk-python/issues/362.

Related slack thread: https://temporalio.slack.com/archives/CTT84RS0P/p1702589499220879 (Issue is described above and the message did not provide more context. Not going to copy-paste the message here)

**Comment by mjameswh** [comment]
Seems like we did in Go. Missing in all other SDKs.

**Comment by cretz** [comment]
Mentioned in Slack, but basically only user-controlled clients can create the initial spans for workflows when starting workflows. We can't know what tracer or tracing framework a user is using to create the span for them when we start the workflow server side. We don't create parentless spans on the worker side if there is no parent span because it would be a bunch of orphans, but maybe we can support orphaned worker-side spans or maybe we can support some new mechanism to attach start-workflow headers on first task. The SDK team will discuss options here.

**Comment by cretz** [comment]
Separate issue opened. This will be fixed in #454.
